python-client/giskard/scanner/calibration/overconfidence_detector.py

```diff
@@ -64,24 +63,72 @@ def _find_issues(
             fail_idx = sliced_dataset.df[(sliced_dataset.df[self.LOSS_COLUMN_NAME] > p_threshold)].index
             relative_delta = (slice_rate - reference_rate) / reference_rate
 
+            # Skip non representative slices
+            # @TODO: do this with a statistical test instead of filtering by count only (GSK-1279)
+            if len(fail_idx) < 20:
+                continue
+
             if relative_delta > self.threshold:
-                level = "major" if relative_delta > 2 * self.threshold else "medium"
-                issues.append(
-                    OverconfidenceIssue(
-                        model,
-                        dataset,
-                        level,
-                        CalibrationIssueInfo(
-                            slice_fn=slice_fn,
-                            slice_size=len(sliced_dataset),
-                            metric_value_slice=slice_rate,
-                            metric_value_reference=reference_rate,
-                            loss_values=meta[self.LOSS_COLUMN_NAME],
-                            fail_idx=fail_idx,
-                            threshold=self.threshold,
-                            p_threshold=p_threshold,
-                        ),
-                    )
+                level = IssueLevel.MAJOR if relative_delta > 2 * self.threshold else IssueLevel.MEDIUM
+                description = (
+                    "For records in your dataset where {slicing_fn}, we found a significantly higher number of "
+                    "overconfident wrong predictions ({num_overconfident_samples} samples, corresponding to "
+                    "{metric_value_perc}% of the wrong predictions in the data slice)."
                 )
+                issue = Issue(
+                    model,
+                    dataset,
+                    group=Overconfidence,
+                    level=level,
+                    description=description,
+                    slicing_fn=slice_fn,
+                    meta={
+                        "metric": "Overconfidence rate",
+                        "metric_value": slice_rate,
+                        "metric_value_perc": slice_rate * 100,
+                        "metric_reference_value": reference_rate,
+                        "num_overconfident_samples": len(fail_idx),
+                        "deviation": f"{relative_delta*100:+.2f}% than global",
+                        "slice_size": len(sliced_dataset),
+                        "threshold": self.threshold,
+                        "p_threshold": p_threshold,
+                    },
+                    tests=_generate_overconfidence_tests,
+                    importance=relative_delta,
+                )
+
+                # Add examples
+                def filter_examples(issue, dataset):
```

Comment:
Because it’s the interface of callbacks in `ExampleExtractor`, which invokes the function with `issue` and `dataset`. In this case `issue` is not used, but it will be passed anyway by the caller.

Reply:
I see, I see, my bad. I skipped the `ExampleExtractor` line apparently.